inst/include/interface/rcpp/rcpp_objects/rcpp_interface_base.hpp

```diff
@@ -200,10 +200,17 @@ class ParameterVector{
   ParameterVector(Rcpp::NumericVector x, size_t size){
     this->id_m = ParameterVector::id_g++;
     this->storage_m = std::make_shared<std::vector<Parameter> >();
-    this->resize(size);
-    for (size_t i = 0; i < size; i++) {
-      storage_m->at(i).initial_value_m = x[i];
-    }
+      this->resize(size);
+      if(x.size() != size){
+          FIMS_WARNING_LOG("Mismatch in constructor ParameterVector(Rcpp::NumericVector x, size_t size). x.size() = " + fims::to_string(x.size())+ " != "+ fims::to_string(size) +". resulting vector will be initialized to " + fims::to_string(x[0]) +" for all values");
+          for (size_t i = 0; i < size; i++) {
+              storage_m->at(i).initial_value_m = x[0];
+          }
```

Comment:
@msupernaw are you sure that we want to do a warning and initialize to zero? Or, should we do a full stop with an error? I think that I prefer the latter but I could be convinced otherwise.

Reply:
@kellijohnson-NOAA no I'm not sure! I'd rather throw an error. I'll make the change!